### DISCUSSION TITLE
Not working with my setup, please help!

### DISCUSSION THREAD

**Discussion by vietnguyen99** [awawa-dev/HyperSerialEsp8266#20] [Q&A]
My setup: 
- Windows 11 PC USB to microUSB of ESP 8266 CH340 
( I flashed my CH340 with HyperSerialEsp8266.ino.d1_mini_WS821x_FIRSTLED_ENABLED.bin) 
- CH340 to WS2812B LEDs. D4 and GND PIN to the green and white respectively.
- 5V power supply to WS2812B LEDs, red to +, white to -.

On HyperHDR, I can see my CH340 detected on COM4. I setup adalights settings and LED layout thru your guide. I don't know why my LED is not turning on. Please help :(

**Comment by awawa-dev** [comment]
Any errors in HyperHDR logs and how you configured your adalight LED driver in HyperHDR? Upload the logs here.

Test procedure for HyperSerialEsp8266 (to check if CH340 is a counterfeited chip)
Close HyperHDR and all programs that could use you COM4. Use Putty and adjust port number to COM4:

![obraz](https://user-images.githubusercontent.com/69086569/200708631-cccd6f28-df83-4e64-aaeb-d9a8d1918599.png)
![obraz](https://user-images.githubusercontent.com/69086569/200708654-cefe394e-b900-402b-b396-e3e9b6e96753.png)

There is newer version of HyperSerialEsp8266 just released, but you should see (give it few seconds) something similar to:

![obraz](https://user-images.githubusercontent.com/69086569/200708681-aed2915f-cf02-4bd9-a3fd-c70278e264a8.png)

**Reply by vietnguyen99** [reply]
![Screenshot 2022-11-08 194754](https://user-images.githubusercontent.com/72474661/200733783-31908ee2-465b-47b2-8ec4-99aa795defca.png)

Does this mean I bought a conterfeit?
Also my adalight driver:
![ada](https://user-images.githubusercontent.com/72474661/200734096-524ae261-6227-4d4a-8b39-e41167c9c33c.png)

My logs:
[logs.txt](https://github.com/awawa-dev/HyperSerialEsp8266/files/9967174/logs.txt)

**Reply by awawa-dev** [reply]
If you set 2000000 in Putty before making a connection, that means that's not a CH340G/C chip but some clone of it. Check markings on your ESP8266 boards, there should be something like 340G/C on the rectangle chip... anyway there is no communication at 2Mb speed and probably that's why your LED doesnt work. 

You can also easily build online a version of HyperSerialEsp8266 for 921600 speed and test it. You can find the instruction on the main page (change ```-DSERIALCOM_SPEED=2000000``` to ```-DSERIALCOM_SPEED=921600``` for your firmware version in ```platformio.ini```). When testing with HyperHDR and Putty you must use 921600 instead 2000000 baud then. Or you can buy a different board. ESP32/HyperSerialESP32 is highly recommended. So far I haven't heard about counterfeited chips CH9102x or CP2104... of course everything could happen and there is minimal risk to receive DOA unit depending on the quality, but we can do nothing about it.

**Reply by vietnguyen99** [reply]
Thank you! I figured it's counterfeit. The chip did not have anything '340' on it. I ordered https://www.amazon.com/dp/B081PX9YFV?ref=ppx_yo2ov_dt_b_product_details&th=1. Hope they are legit.

**Reply by awawa-dev** [reply]
Usually the problematic chips on ESP8266 don't have any markings and sometimes even they lack support of basic hardware features (instead they rely on the driver software emulation which makes things even worse). Nevertheless most of them can still work at @921600 speed so you can at least test your LED strip now.

**Reply by vietnguyen99** [reply]
I tested with 921600 baud and it works! In the mean time, I will use this fake CH340 while waiting for my CH9102. Thank you!

**Comment by appleimperio** [comment]
I been using HyperHDR with WLED over wifi for a long time and it was working perfect. I decide to try the HyperSerial8266 but unfortunately i bought the same board with counterfeit chip so I have to lower the speed as awawa suggest for it to work. Is there any difference in speed between the downgraded 921600 and the WLED over wifi? I just want to know in the meantime while i wait for the new boards if going back to WLED or stay with the HyperSerial at 921600.

**Reply by awawa-dev** [reply]
Hard to assess without knowing more details about your setup...how many LEDs and your ping from device hosting HyperHDR to Esp8266 running WLED (don't know how to measure the wifi speed): requires a lot of testing because even if one is over 100ms your will notice it using WLED and that never will happen with HyperSerialEsp8266.

**Reply by appleimperio** [reply]
222 LED and the ping min/avg/max/mdev = 1.757/5.547/28.389/5.474 ms

**Reply by awawa-dev** [reply]
Considering only the serial port speed:
```
for  921600 speed 222 LEDS => 222*3 bytes => 222*3*8 bits => 5328/ 921600 => additional 5.78ms transport latency 
for 2Mb speed 222 LEDS => 222*3 bytes => 222*3*8 bits => 5328/ 2000000 => additional 2.66ms transport latency
```

**Reply by appleimperio** [reply]
interesting. Thank you!
